File: javascript-solutions/01-contains-duplicate/solution-two.js
//Given an integer array nums, return true if any value appears more than once in the array, otherwise return false.

// Time & Space complexity O(n) ✅

class Solution {

    hasDuplicate(nums) {
        return new Set(nums).size < nums.length;
    }
}


const test1 = [1, 2, 3 ,4];
const test2 = [1, 2, 3, 1];
const test3 = [];
const test4 = [5, 5, 5, 5];

const solution = new Solution();    // Create a new instance of Solution

console.log(solution.hasDuplicate(test1));      // false
console.log(solution.hasDuplicate(test2));      // true
console.log(solution.hasDuplicate(test3));      // false
console.log(solution.hasDuplicate(test4));      // true

// or input directly

console.log(solution.hasDuplicate([1, 2, 3, 4]));
console.log(solution.hasDuplicate([1, 2, 3, 1]));

File: java-solutions/01-contains-duplicate/SolutionTwo.java
//Given an integer array nums, return true if any value appears more than once in the array, otherwise return false.

import java.util.Arrays;

// Time & Space complexity (O(n)) ✅ 

public class SolutionTwo {

    public boolean containsDuplicate(int[] nums) {

        return Arrays.stream(nums).distinct().count() < nums.length;
    }

    public static void main(String[] args) {

        SolutionTwo sol = new SolutionTwo();

        int[] test1 = {1, 2, 3, 4};
        int[] test2 = {1, 2, 3, 1};
        int[] test3 = {};
        int[] test4 = {5, 5, 5, 5};

        System.out.println(sol.containsDuplicate(test1));
        System.out.println(sol.containsDuplicate(test2));
        System.out.println(sol.containsDuplicate(test3));
        System.out.println(sol.containsDuplicate(test4));

    }
}

                                                                    